idaes/core/util/tables.py

```diff
@@ -266,22 +268,50 @@ def stream_table_dataframe_to_string(stream_table, **kwargs):
         na_rep=na_rep, justify=justify, float_format=float_format, **kwargs
     )
 
-
-def _get_state_from_port(port, time_point):
-    # Check port for _state_block attribute
-    try:
-        if len(port._state_block) == 1:
-            return port._state_block[0][time_point]
-        else:
-            return port._state_block[0][time_point, port._state_block[1]]
-    except AttributeError:
-        # Port was not created by IDAES add_port methods. Return exception for
-        # the user to fix.
-        raise ConfigurationError(
-            f"Port {port.name} does not have a _state_block attribute, "
-            f"thus cannot determine StateBlock to use for collecting data."
-            f" Please provide the associated StateBlock instead, or use "
-            f"the IDAES add_port methods to create the Port."
+def _get_state_from_port(port,time_point):
+    from idaes.core.property_base import StateBlockData
+    states = list()
+    v = None # To make PyLint stop complaining
+    for v in port.iter_vars():
+        if not v.parent_block().parent_component().is_indexed():
+            raise TypeError(
+                f"No state block could be retrieved from Port {port.name} "
+                f"because variable {v.name} does not belong to a state block."
+                )
+        if not isinstance(v.parent_block(),StateBlockData):
+            raise TypeError(
+                f"No state block could be retrieved from Port {port.name} "
+                f"because variable {v.name} does not belong to a state block."
+                )
+        states.append(v.parent_block().parent_component())
+        
+    if len(states) == 0:
+        raise AttributeError(
+            f"No state block could be retrieved from Port {port.name} "
+            f"because it contains no variables."
+            )
+    # Check the number of indices of the parent property block. If its indexed
+    # both in space and time, keep the second, spatial index and throw out the
+    # first, temporal index. If that ordering is changed, this method will
+    # need to be changed as well.
+    idx = v.parent_block().index()
```

Comment:
I think you should generally not use the for loop variable outside the loop.  As an alternative, and to make pylint happy you could do (I assume all states are the same):

```
idx = state[0].index()
```

Reply:
That doesn't make this good practice thought. You do not directly control `v` so you don't know for sure what it is is. You also needed to add a hack `v = None` initially to satisfy pylint which is a good indication you should do something differently (i.e. you should fix the problem, not just throw something in to hide the error).